### ISSUE TITLE
[Bug]: Virtual id limit in OTG of 1024 & WE repo outdated.

### ISSUE THREAD

**Issue by NLBlackEagle** [PG85/OpenTerrainGenerator#984]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

TL;DR: This whole report is just backtracing steps where I already fixed the issue and am about to do a pull request. This simply describes the issue I found and the steps I took to find the core issue after which I fixed it.

In ChunkProviderOTG.java there is a limit set to the amount of virtual id's a preset can have. This is causing some issues where 1024 - 256 reserved slots results in having a maximum of 768 slots for virtual ID's. If this is exceeded Minecraft will crash 

This bug was found whilst having a bunch of mods installed as dependencies with my RLCraft version of Dregora and as such the crash report will differ with different mods.

`    at net.minecraft.world.gen.structure.MapGenStructure.func_180701_a(MapGenStructure.java:38)
    at net.minecraft.world.gen.MapGenBase.func_186125_a(SourceFile:34)
    at ivorius.reccomplex.world.gen.feature.structure.MapGenStructureHook.func_186125_a(MapGenStructureHook.java:153)
    at ivorius.reccomplex.events.handlers.RCTerrainGenEventHandler$5.func_186125_a(RCTerrainGenEventHandler.java:329)
    at com.pg85.otg.forge.world.ForgeWorld.prepareDefaultStructures(ForgeWorld.java:945)
    at com.pg85.otg.generator.ChunkProviderOTG.generate(ChunkProviderOTG.java:132)
    at com.pg85.otg.forge.generator.OTGChunkGenerator.getBlocks(OTGChunkGenerator.java:259)
    at com.pg85.otg.forge.generator.OTGChunkGenerator.func_185932_a(OTGChunkGenerator.java:129)`
    
After searching for the class points to (Quote from PG85) "OTG's vanilla structures inherit from MapGenStructure instead of their respective classes causing other mods to trip up when Forge events are fired." as said here: https://github.com/PG85/OpenTerrainGenerator/issues/352

### Expected Behavior

Id limit in ChunkProviderOTG.java to be increased to 2048 and fix the issue where minecraft would crash if ID limit is exceeded.

### Which platforms have you reproduced this issue using?

Forge Single Player

### OTG Version

OTG 1.12.2 V9.4 R1

### Platform Version

Forge 14.23.5.2860

### Steps To Reproduce

Already fixed it so there's probably no need for this.

### Log file

https://pastebin.com/ie9bv0Bq

**Comment by NLBlackEagle** [comment]
Issue Fixed
